src/vs/workbench/contrib/search/browser/searchActionsTextQuickAccess.ts

```diff
@@ -29,6 +34,28 @@ registerAction2(class TextSearchQuickAccessAction extends Action2 {
 
 	override async run(accessor: ServicesAccessor, match: RenderableMatch | undefined): Promise<any> {
 		const quickInputService = accessor.get(IQuickInputService);
-		quickInputService.quickAccess.show(TEXT_SEARCH_QUICK_ACCESS_PREFIX);
+		const searchText = getSearchText(accessor) ?? '';
+		quickInputService.quickAccess.show(TEXT_SEARCH_QUICK_ACCESS_PREFIX + searchText);
 	}
 });
+
+function getSearchText(accessor: ServicesAccessor): string | null {
+	const editorService = accessor.get(IEditorService);
+	const configurationService = accessor.get(IConfigurationService);
+
+	const activeEditor: IEditor = editorService.activeTextEditorControl as IEditor;
+	if (!activeEditor) {
+		return null;
+	}
+	if (!activeEditor.hasTextFocus()) {
+		return null;
+	}
+
+	// only happen if it would also happen for the search view
+	const seedSearchStringFromSelection = configurationService.getValue<IEditorOptions>('editor').find!.seedSearchStringFromSelection;
```

Comment:
I believe this can be simplified to this without having to use the `!`.
```ts
	const seedSearchStringFromSelection = configurationService.getValue<boolean>('editor.find.seedSearchStringFromSelection');
```

Reply:
good idea